qiskit_ibm_runtime/runtime_job_v2.py

```diff
@@ -0,0 +1,788 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qiskit runtime job."""
+
+from typing import Any, Optional, Callable, Dict, Type, Union, Sequence, List, Literal, Tuple
+import time
+import json
+import logging
+from concurrent import futures
+import traceback
+import queue
+from datetime import datetime
+import requests
+
+from qiskit.providers.backend import Backend
+from qiskit.providers.models import BackendProperties
+from qiskit.primitives.base.base_primitive_job import BasePrimitiveJob
+
+# pylint: disable=unused-import,cyclic-import
+
+from qiskit_ibm_runtime import qiskit_runtime_service
+
+from .utils import utc_to_local
+from .utils.utils import validate_job_tags
+from .utils.estimator_result_decoder import EstimatorResultDecoder
+from .utils.queueinfo import QueueInfo
+from .constants import API_TO_JOB_ERROR_MESSAGE, DEFAULT_DECODERS
+from .exceptions import (
+    IBMApiError,
+    RuntimeJobFailureError,
+    RuntimeInvalidStateError,
+    IBMRuntimeError,
+    RuntimeJobTimeoutError,
+    RuntimeJobMaxTimeoutError,
+)
+from .utils.result_decoder import ResultDecoder
+from .api.clients import RuntimeClient, RuntimeWebsocketClient, WebsocketClientCloseCode
+from .exceptions import IBMError
+from .api.exceptions import RequestsApiError
+from .api.client_parameters import ClientParameters
+
+logger = logging.getLogger(__name__)
+
+JobStatus = Literal["INITIALIZING", "QUEUED", "RUNNING", "CANCELLED", "DONE", "ERROR"]
+
+JOB_FINAL_STATES: Tuple[JobStatus, ...] = ("DONE", "CANCELLED", "ERROR")
+
+API_TO_JOB_STATUS: Dict[str, JobStatus] = {
+    "QUEUED": "QUEUED",
+    "RUNNING": "RUNNING",
+    "COMPLETED": "DONE",
+    "FAILED": "ERROR",
+    "CANCELLED": "CANCELLED",
+}
+
+
+class RuntimeJobV2(BasePrimitiveJob):
```

Comment:
Instead of duplicating all the same methods, why not just create a `BaseRuntimeJob` that has all the common method. Then `RuntimeJobV2` can inherit from both `BaseRuntimeJob` and `BasePrimitiveJob`.